### PR TITLE
[CI:DOCS]First pass at release process

### PR DESCRIPTION
**Pull request by cevich**
Import the release process docs from a private google-doc for accessibility and ease of maintenance.

### CODE REVIEW

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      Please seek assistance if needed, before proceeding. Do not merge without all automated testing, including
+      the release test, passing.
+   1. Once this PR merges, push the tag with `git push upstream vX.Y.Z`
+
+1. Verify automated build and actual release-testing passes by going to:
+   * `https://cirrus-ci.com/github/containers/podman/<vX.Y.Z>`
+   * If there is a failure, be 100% sure you understand and accept the cause, given any code changes will require a new tag.
+
+1. Bump master to new -dev version
+   1. If you made a release branch and bumped major or minor version, complete steps 10 to 12 again but on the master
```
Comment (cevich) [inline RELEASE_PROCESS.md:61]: Matt suggested:

    The "Steps 10 to 12" bit probably needs to be adjusted

File: RELEASE_PROCESS.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+* For a major release, you have 4-8 hours of time available, most of which will
+  be dedicatd to writing release notes.
+* For a minor or micro release, you have 2-4 hours of time available (minimum depends
+  largely on the speed/reliability of automated testing)
+
+# Release Process
+
+## Overview
+
+## Micro version (`*.*.Z`)
```
Comment (vrothberg) [inline RELEASE_PROCESS.md:22]: [semver](https://semver.org/) refers to that as the "PATH" version.

**Review by vrothberg — COMMENTED** [review]
I'd like a `s/micro/patch` as it's a established term in semver.  Other than that LGTM

**Comment by cevich** [comment]
Oh, my bad.  This is no where near ready for review :frowning: Sorry if I wasted anybody's time.

> I'd like a s/micro/patch as it's a established term in semver. Other than that LGTM

I agree, "patch" is much more meaningful than "micro".

File: RELEASE_PROCESS.md (hunk 2 above)
Comment (cevich) [reply RELEASE_PROCESS.md:22]: yes, I agree "patch" is better.  I'll fix it.

**Review by vrothberg — COMMENTED** [review]
LGTM

File: RELEASE_PROCESS.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+## Release workflow expectations
+
+* You have push access to the [upstream podman repository][https://github.com/containers/podman.git]
+* You understand all basic `git` operations and concepts, like creating commits,
+  local vs. remote branches, rebasing, and conflict resolution.
+* You have access to your public and private *GPG* keys.
+* You have reliable internet access (i.e. not the public WiFi link at McDonalds)
+* Other podman maintainers are online/available for assistance if needed.
+* For a **major** release, you have 4-8 hours of time available, most of which will
+  be dedicatd to writing release notes.
```
Comment (lsm5) [inline RELEASE_PROCESS.md:22]: typo

Comment (cevich) [reply RELEASE_PROCESS.md:22]: Got it, thanks.  I'm not going to push yet, since there are other updates I need to make.

**Comment by cevich** [comment]
@containers/podman-maintainers PTAL draft, or at least glace through it.  Sorry it's **HUGE** and it's **HAIRY** (with some bonus monster drool).  I'm sure it's not perfect and there are mistakes and missing links.  I'd appreciate feedback on any glaring mistakes/omissions - Then I'll make another pass or three over it and refine further as required.  Thanks :grin:

File: RELEASE_PROCESS.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+      failures.  This can be done by going directly to
+      `https://cirrus-ci.com/github/containers/podman/vX.Y`
+   1. If there are CI test or automation boops that need fixing on the branch,
+      attend to them using normal PR process (to *master* first, then backport
+      changes to the new branch).  Ideally, CI should be "green" on the new
+      branch before proceeding.
+
+1. Create a new branch-verification Cirrus-Cron entry.
+
+  1. This is to ensure CI's VM image timestamps are refreshed.  Without this,
```
Comment (ashley-cui) [inline RELEASE_PROCESS.md:137]: Misalignment of bullet point by a single space :) It's causing it to not be sub points
```suggestion
   1. This is to ensure CI's VM image timestamps are refreshed.  Without this,
```

File: RELEASE_PROCESS.md [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+
+1. Make a `[CI:DOCS]` release notes pull request.
+
+   1. Ensure any/all intended PR's are completed and merged prior to any
+      processing of release notes.  Ensure your local clone is fully up to
+      date with the remote upstream (`git remote update`).
+   1. Check out (create) a local working branch for a release-notes PR,
+      based on the latest `upstream/master` or pre-existing version-named
+      branch - for example, if this is an additional *release-candidate*
+      (e.g. x.y.z-rc2).
```
Comment (ashley-cui) [inline RELEASE_PROCESS.md:66]:
```suggestion
      (e.g. vX.Y.Z-rc2).
```

File: RELEASE_PROCESS.md [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+
+# Releases
+
+## Major (***X***.y.z) release
+
+These releases always begin from *master*, and are contained in a branch
+named with the **major** and **minor** version. **Major** release branches
+begin in a *release candidate* phase, with prospective release tags being
+created with an `-rc` suffix.  There may be multiple *release candidate*
+tags before the final/official **Major** version is tagged and released.
```
Comment (ashley-cui) [inline RELEASE_PROCESS.md:34]:
```suggestion
tags before the final/official **major** version is tagged and released.
```

File: RELEASE_PROCESS.md [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+standards.
+Significant versions
+are tagged, including *release candidates* (`rc`).
+All relevant **major** releases have their own branches.  The **latest**
+development efforts occur on the *master* branch.  Branches with a
+*rhel* suffix are use for long-term support of downstream RHEL releases.
+
+## Release workflow expectations
+
+* You have push access to the [upstream podman repository][https://github.com/containers/podman.git]
```
Comment (ashley-cui) [inline RELEASE_PROCESS.md:15]:
```suggestion
* You have push access to the [upstream podman repository](https://github.com/containers/podman.git)
```

File: RELEASE_PROCESS.md [hunk 8]
```diff
@@ -0,0 +1,268 @@
+# Podman Releases
+
+## Overview
+
+Podman (and podman-remote) versioning is mostly based on semantic-versioning
```
Comment (ashley-cui) [inline RELEASE_PROCESS.md:5]: Should we link to what semantic versioning is? 
```suggestion
Podman (and podman-remote) versioning is mostly based on [semantic-versioning](https://semver.org/)
```

**Review by ashley-cui — COMMENTED** [review]
a bunch of small nits :)

File: RELEASE_PROCESS.md [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+standards.
+Significant versions
+are tagged, including *release candidates* (`rc`).
+All relevant **major** releases have their own branches.  The **latest**
```
Comment (mheon) [inline RELEASE_PROCESS.md:9]: Technically it's relevant minor releases - 2.0 and 2.1 and 2.2 were all on separate upstream branches.

File: RELEASE_PROCESS.md (hunk 8 above)
Comment (vrothberg) [reply RELEASE_PROCESS.md:5]: Good idea

**Review by vrothberg — COMMENTED** [review]
@cevich,  could you add that `major` releases require to bump the go module in `go.mod`?

Let's get this PR in.

**Comment by cevich** [comment]
> @cevich, could you add that major releases require to bump the go module in go.mod?
> Let's get this PR in.

Sure.  I was leaving it sit to gather feedback for a final draft update, I'll likely get to that next week. I'm currently consumed with getting F34b VM Images building smoothly.

**Comment by cevich** [comment]
@containers/podman-maintainers PTAL final draft ready for review/merge.

**Review by rhatdan — APPROVED** [review]
/lgtm

**Pull request merged**